File: src/main/java/personal/pkonigstein/datatypes/BingoCell.java
package personal.pkonigstein.datatypes;

import java.text.SimpleDateFormat;
import java.util.Date;

public class BingoCell extends DataEntry{

    private static long idCounter= 0;
    private boolean happened;
    private long id;

    private String anyoneWho = "";

    private String happenedTime;

    public BingoCell() {
    }

    public BingoCell(DataEntry entry){
        this.id = idCounter++;
        this.action = entry.action;
        this.name = entry.name;
        this.happened = false;
        this.anyoneWho = "";
    }

    public BingoCell(String name, String action) {
        super(name, action);
        this.id = idCounter++;
        this.happened = false;
        this.anyoneWho = "";
    }

    public boolean isHappened() {
        return happened;
    }

    public void setHappened(boolean happened) {
        this.happened = happened;
        this.happenedTime = new SimpleDateFormat("HH:mm:ss").format(new Date());
    }

    public long getId() {
        return id;
    }

    public String getHappenedTime() {
        return happenedTime;
    }

    public String getAnyoneWho() {
        return anyoneWho;
    }
}

File: src/mensabingo/src/datatypes/BIngoCell.ts
export {BingoCell, BingoField}
class BingoCell {

  name: string;
  action: string;
  happened : boolean;

  constructor(name?: string, action?: string, happened?: boolean) {
    if(name){
      this.name = name;
    }
    else{
      this.name ="";
    }
    if(action){
      this.action = action;
    }
    else{
      this.action = "";
    }
    if(happened){
      this.happened = happened;
    }
    else{
      this.happened = false;
    }
  }


}

class BingoField {
  field : BingoCell[][];

  constructor() {
    this.field = [];
    for(let i=0;i<5;i++){
      let row:BingoCell[] = []
      for(let j=0;j<5;j++){
        row.push(new BingoCell());
      }
      this.field.push(row);
    }
  }
}
